### PR TITLE
Enable alerting

### PR DESCRIPTION
**Pull request by EMaksy**
This PR will enable/disable the alerting configuration (by using env variables) in Ansible playbook

Alerting is disabled by default. By providing a json configuration file when running the playbook, the user is able to enable alerting. 

When disabled, the smtp configuration variables are not used for the web container. 

Added: 
- json config example in README
- env description in README
- default env values for alerting
- optional env when alerting is enabled

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,25 @@ all:
 }
 ```
 
+**Example json variables  to install trento-server with the all in one node configuration and enable alerting**
```
Comment (arbulu89) [inline README.md:88]: Double space after `variables` word

File: README.md [hunk 2]
```diff
@@ -285,6 +303,12 @@ These variables are the defaults of our roles, if you want to override the prope
 | refresh_token_secret | The secret used for refresh tokens jwt signature | Generated by playbook |
 | web_admin_username | Username of the admin user in web application | admin |
 | enable_alerting | Enable the alerting mechanism on web project | false |
+| alert_sender     | Email address used as the "from" address in alerts | alerts@trento-project.io  |
```
Comment (arbulu89) [inline README.md:306]: could we format this like the other rows? only letting one space after the values

File: README.md [hunk 3]
```diff
@@ -285,6 +303,12 @@ These variables are the defaults of our roles, if you want to override the prope
...
+| alert_recipient  | Email address to receive alert notifications | admin@trento-project.io   |
+| smtp_server      | IP address of the SMTP server |   |
+| smtp_port        | Port number of SMTP server |   |
+| smtp_user        | Username for SMTP authentication. |   |
```
Comment (arbulu89) [inline README.md:310]: Let's remove that `.` after authentication

File: roles/containers/defaults/main.yml [hunk 4]
```diff
@@ -25,8 +25,14 @@ secret_key_base: "{{ lookup('community.general.random_string', base64=True, leng
...
-enable_alerting: "false"
 enable_api_key: "true"
 enable_charts: "true"
+enable_alerting: "false"
+smtp_server: ""
+smtp_port: ""
+smtp_user: ""
+smtp_password: ""
+alert_sender: "alerts@trento-project.io"
+alert_recipient: "admin@trento-project.io"
```
Comment (arbulu89) [inline roles/containers/defaults/main.yml:36]: I wouldn't put such a default. I would let it empty. I don't think nobody has `admin@trento-project.io` as their mail

File: roles/containers/tasks/main.yml [hunk 5]
```diff
@@ -47,6 +47,35 @@
       AMQP_URL: "{{ amqp_protocol }}://{{ rabbitmq_username }}:{{ rabbitmq_password }}@{{ rabbitmq_host }}/{{ rabbitmq_vhost | urlencode | replace('/', '%2F') }}"
       DATABASE_URL: "ecto://{{ wanda_postgres_user }}:{{ wanda_postgres_password }}@{{ wanda_postgres_host }}/{{ wanda_postgres_db }}"
 
+- name: Prepare env variables for Web container
+  ansible.builtin.set_fact:
```
Comment (arbulu89) [inline roles/containers/tasks/main.yml:51]: I'm not sure at all if setting this values as facts is a good idea.
When you set values as fact, they can be retrieved later on from the main ansible host machine, etc.
I mean, they have their usage. Here, you are using them as simple value holders.
Personally, I would put them all directly in the container env entry.

I know that you did put some effort on this, but I don't think we should use it.

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hey @EMaksy 
Good effort.
I think I would "revert" how the envs are used. Good work otherwise

File: roles/containers/defaults/main.yml (hunk 4 above)
Comment (EMaksy) [reply roles/containers/defaults/main.yml:36]: Sure we can let them empty!
I was just reusing the same default values we use in [trento web configs ](https://github.com/trento-project/web/blob/main/config/config.exs#L35):+1:

File: roles/containers/tasks/main.yml (hunk 5 above)
Comment (EMaksy) [reply roles/containers/tasks/main.yml]: Sure, I just wanted to skip passing empty env's to the docker container. 
I reverted the changes :+1:

**Review by CDimonaco — CHANGES_REQUESTED** [review]
Good job, really but I think we are missing a point

This variables are dependent and we should avoid errors from the user and configuration in general

I think we should have a pre task, to check for this condition

if the alerting is enabled, the alerting config variables can't be empty

Maybe we are missing this type of checks also for other features, but the alerting feature is quite different, you could have alerting enabled, with empty config variables and you can just found out later the error without having the application exploding immediately

A task like that https://stackoverflow.com/questions/45013306/how-to-make-ansible-variable-mandatory/61563967#61563967

**Review by CDimonaco — APPROVED** [review]

**Pull request merged**